sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LegacyServiceClient.cs

```diff
@@ -384,6 +386,8 @@ private Response<RecognizeEntitiesResultCollection> RecognizeEntitiesBatch(Multi
         public override async Task<Response<PiiEntityCollection>> RecognizePiiEntitiesAsync(string document, string language = default, RecognizePiiEntitiesOptions options = default, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(document, nameof(document));
+            Validation.SupportsOperation(nameof(TextAnalyticsClient.RecognizePiiEntitiesAsync), TextAnalyticsClientOptions.ServiceVersion.V3_1, ServiceVersion);
```

Comment:
I would also consider post-facto validation in this case.  It could go in the `catch` block below.

Reply:
As noted above, the service error is not specific enough and the version check itself is minimal - no different than all our argument checks. If the perf of this is concerning, we should do more to elide argument checks in internal/private code and do that all in public code so it's not repeated.